### PR TITLE
Cleaned the repo and now is used the original solita

### PR DESCRIPTION
**Pull request by ricardocr987**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -21,7 +22,10 @@ async function main() {
     let path: string[] = options.file.replace('.json', '').split('/')
     let programName: string = path[path.length - 1]
     const paths = new Paths(`./`, programName)
-    const idl = JSON.parse(readFileSync(paths.idlFile(programName), "utf8"))
+    const idl: Idl = JSON.parse(readFileSync(paths.idlFile(programName), "utf8"))
+    idl.metadata = {
+      address: "PROGRAM PUBKEY"
```
Comment (MHHukiewitz) [inline src/index.ts:27]: won't this overwrite the address, in case that it is actually provided in the JSON?

File: src/render-domain.ts [hunk 2]
```diff
@@ -306,28 +307,37 @@ export default class MainDomain
...
-      globalStats.totalRequests += totalRequests
-      globalStats.totalUniqueAccessingPrograms += totalUniqueAccessingPrograms
+      globalStats.totalAccounts[type]++
+      globalStats.totalAccesses += accesses
+      Object.entries(accessesByProgramId).forEach(([programId, accesses]) => {
+        globalStats.totalAccessesByProgramId[programId] =
+          (globalStats.totalAccessesByProgramId[programId] || 0) + accesses
+      })
+      globalStats.startTimestamp = Math.min(
+        globalStats.startTimestamp || Infinity, startTimestamp || Infinity,
```
Comment (MHHukiewitz) [inline src/render-domain.ts:328]:
```suggestion
        globalStats.startTimestamp || Number.MAX_SAFE_INTEGER, startTimestamp || Number.MAX_SAFE_INTEGER,
```
Needed to replace Infinity with MAX_SAFE_INTEGER

File: src/types.ts [hunk 3]
```diff
@@ -183,7 +183,6 @@ export type IdlType =
   | "i64"
   | "u128"
   | "i128"
-  | "bytes"
```
Comment (MHHukiewitz) [inline src/types.ts:186]: why remove?

**Review by MHHukiewitz — APPROVED** [review]
LGTM. Might need further updates as the marinade indexer gets fixed this week

**Comment by MHHukiewitz** [comment]
I just generated the Marinade indexer. When generating the solita code, you now provide the programId pubkey to the beet objects:
```ts
export function createAddLiquidityInstruction(
  accounts: AddLiquidityInstructionAccounts,
  args: AddLiquidityInstructionArgs,
  programId = new web3.PublicKey('PROGRAM PUBKEY'),
) 
```
but you could refer to constants.js here instead.

**Comment by ricardocr987** [comment]
> I just generated the Marinade indexer. When generating the solita code, you now provide the programId pubkey to the beet objects:
> 
> ```ts
> export function createAddLiquidityInstruction(
>   accounts: AddLiquidityInstructionAccounts,
>   args: AddLiquidityInstructionArgs,
>   programId = new web3.PublicKey('PROGRAM PUBKEY'),
> ) 
> ```
> 
> but you could refer to constants.js here instead.

This is generated by solita (which we dont have control). Also, this happens when you provide the idl file, not the pubkey to fetch the idl, we could add another argument on the idl file option to include the pubkey to fill that field of the solita IDL type definition.

Anyway, this is a function to create instructions, which we are not going to use.

**Pull request merged**